### PR TITLE
ES6 refactor

### PR DESCRIPTION
**Pull request by tstirrat**
Refactor it all into es6 modules :)

#120

### CODE REVIEW

**Comment by tstirrat** [comment]
Thanks @MattMSumner for your help

**Comment by MattMSumner** [comment]
@tstirrat no worries. Glad to help

File: addon/adapters/firebase.js [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+    var recordCache = Ember.get(adapter._recordCacheForType, fmt('%@.%@', [type.typeKey, record.get('id')])) || {};
+
+    var serializedRecord = record.serialize({includeId:false});
+
+    return new Promise(function(resolve, reject) {
+      var savedRelationships = Ember.A();
+      record.eachRelationship(function(key, relationship) {
+        var save;
+        if (relationship.kind === 'hasMany') {
+            if (serializedRecord[key]) {
```
Comment (MattMSumner) [inline addon/adapters/firebase.js]: indentation is off?

File: addon/adapters/firebase.js [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+              save = adapter._saveHasManyRelationship(store, type, relationship, serializedRecord[key], recordRef, recordCache);
+              savedRelationships.push(save);
+              // Remove the relationship from the serializedRecord because otherwise we would clobber the entire hasMany
+              delete serializedRecord[key];
+            }
+        } else {
+            if (relationship.options.embedded === true && serializedRecord[key]) {
```
Comment (MattMSumner) [inline addon/adapters/firebase.js]: same here?

**Comment by MattMSumner** [comment]
Still getting to grips with the code but this seems like an awesome refactor to me :+1:

File: addon/adapters/firebase.js [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+    Call _saveHasManyRelationshipRecord on each record in the relationship
+    and then resolve once they have all settled
+  */
+  _saveHasManyRelationship: function(store, type, relationship, ids, recordRef, recordCache) {
+    if (!Ember.isArray(ids)) {
+      throw new Error('hasMany relationships must must be an array');
+    }
+    var adapter = this;
+    var idsCache = Ember.A(recordCache[relationship.key]);
+    ids = Ember.A(ids);
```
Comment (stefanpenner) [inline addon/adapters/firebase.js]: rather then doing this, just use EnumerableUtils to get filter.

Comment (tstirrat) [inline addon/adapters/firebase.js]: due to the size of the refactor, I don't want to do much alteration to the adapter/serializer code here. I'd prefer to do the next stage of alteration once we have this merged into master. Seems like  Matt created this change as a new PR anyway.

Comment (stefanpenner) [inline addon/adapters/firebase.js]: sounds good.
I only say the issue when skimming the PR, so i figured i'll raise attention to it. But it should not prevent this PR from merging. As this was pre-existing.

File: bower.json [hunk 4]
```diff
@@ -20,17 +20,13 @@
...
-    "tasks",
+    "tests",
     "vendor",
-    "examples",
     "node_modules",
     "bower_components",
     "build.sh",
     "changelog.txt",
-    "Gruntfile.js",
+    "Brocfile.js",
```
Comment (jwngr) [inline bower.json]: Also include the gulpfile.

File: gulpfile.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+gulp.task('build-legacy', function() {
+  return gulp.src('vendor/legacy-shims/emberfire.js')
+    .pipe(sourcemaps.init())
+    .pipe(transpile({
+      importPaths: ['vendor/legacy-shims'],
+      formatter: 'bundle'
+    }))
+    .pipe(concat('emberfire.js'))
+    .pipe(sourcemaps.write('./'))
```
Comment (jwngr) [inline gulpfile.js]: Add linting back into here.

File: vendor/legacy-shims/emberfire.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+/*!
+ * EmberFire is the officially supported adapter for using Firebase with
+ * Ember Data. The DS.FirebaseAdapter provides all of the standard DS.Adapter
+ * methods and will automatically synchronize the store with Firebase.
+ *
+ * EmberFire 0.0.0
```
Comment (jwngr) [inline vendor/legacy-shims/emberfire.js]: Add to Catapult.

Comment (tstirrat) [inline vendor/legacy-shims/emberfire.js]: done

**Comment by sararob** [comment]
@tstirrat nice work!

**Pull request merged**